packages/pwa/app/utils/responsive-image.js

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {getAppOrigin} from 'pwa-kit-react-sdk/utils/url'
+
+/**
+ * @param {Object} props
+ * @param {string} props.src - The image's url
+ * @param {Object} [props.sizes] - Defines how the image responds across Chakra breakpoints
+ * @param {number[]} [props.widths] - Will be used to generate the image's srcset
+ * @param {Object} [props.transformations] - How to transform the image's url {@link https://documentation.b2c.commercecloud.salesforce.com/DOC1/topic/com.demandware.dochelp/content/b2c_commerce/topics/image_management/b2c_creating_image_transformation_urls.html}
+ * @return {Object} - Props for an image component
+ *
+ * @example
+ * <Img
+ *     {...getImageProps({
+ *         src,
+ *         sizes: {base: 'calc(100vw / 2)', md: '500px'},
+ *         widths: [500, 1000, 1500],
+ *         transformations: {q: 60}
+ *     })}
+ * />
+ */
+export const getImageProps = ({src, sizes, widths, transformations, ...otherProps}) => {
+    const url = transformImageUrl(src, transformations)
```

Comment:
I'm not sure how I feel about this average logic. Is the additional complexity worth the end result? I'm not sure, my first assumption would be that the default might have no transformations at all. In terms of defining your jsdocs, saying "src will be used as the fallback url if your browser doesn't support image sets" is a lot easier than "the default image will be the average width of the values passed in the widths array if you browser doesn't support image sets". 

Also, this calculation would probably have a negative impact on caching if I was to hazard a guess.